api/src/service/domain/workflow/project_history_get.ts

```diff
@@ -1,19 +1,18 @@
 import Intent from "../../../authz/intents";
 import { Ctx } from "../../../lib/ctx";
-import logger from "../../../lib/logger";
 import * as Result from "../../../result";
+import { NotAuthorized } from "../errors/not_authorized";
 import { NotFound } from "../errors/not_found";
 import { Identity } from "../organization/identity";
 import { ServiceUser } from "../organization/service_user";
 import * as Project from "./project";
 import { ProjectTraceEvent } from "./project_trace_event";
-import { NotAuthorized } from "../errors/not_authorized";
 
 export interface Filter {
-  publisher: Identity;
-  startAt: string; // ISO timestamp
-  endAt: string; // ISO timestamp
-  eventType: Identity;
+  publisher?: Identity;
+  startAt?: string; // ISO timestamp;
+  endAt?: string; // ISO timestamp;
+  eventType?: string;
```

Comment:
why has been the "?" operator added?

Reply:
I've done the "filter my not be a empty object" for project.viewhistory.